src/crimson/os/seastore/seastore.cc

```diff
@@ -1081,7 +1081,7 @@ SeaStore::_omap_set_kvs(
         !root.is_null() ?
         tm_iertr::now() :
         omap_manager.initialize_omap(
-          t, onode->get_hint()
+          t, onode->get_hint() + OMAP_LADDR_SPACE
```

Comment:
Why not modify get_hint() to generate a random hint within the onode's range?  That would remove the need for the omap specific logic.

Reply:
I think object data writes also rely on `Onode::get_hint()`, if we randomize `Onode::get_hint()`, it might lead to laddr conflicts between object data extents and omap extents, am I right?

Reply:
Objects tend not to use both the data payload and omap payload heavily, generally one or the other.  Also, if we're randomizing the addresses within that space, even if both are in use, shouldn't it still be ok?

Reply:
> Objects tend not to use both the data payload and omap payload heavily, generally one or the other. 

Yes, I totally agree.

> Also, if we're randomizing the addresses within that space, even if both are in use, shouldn't it still be ok?

It seems that the current object data handler tries to reserve objects' whole laddr space at the time of the first write. And I think we probably will want to set objects' size to 'GB' level when crimson is used in production environments, because that will reduce the number of objects and give us the possibility to cache all objects' metadata in memory, which would dramatically increase system performance as no metadata related I/O is needed for data reads/writes. So, say we set objects' size to 1 GB and reserve its whold laddr space at the time of the first data write, if we randomize `Onode::get_hint()`, later omap extent alloc would have a high chance to collide with that 1 GB data laddr space reservation. Am I right?:-)

Reply:
object_data_handler only reserves 16MB.  I don't want to worry about GB size objects at this time -- there are a lot of other things we'd need to change for that to be a good idea.

Reply:
I think simply randomizing the hint will get us the same performance advantage at this time with much less complexity.

Reply:
will do:-)

Reply:
done:-)